kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/TopicsResource.java

```diff
@@ -132,6 +134,36 @@ public void getTopic(
     AsyncResponses.asyncResume(asyncResponse, response);
   }
 
+  @PATCH
+  @Path("/{topicName}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @PerformanceMetric("v3.topics.partitions")
+  @ResourceName("api.v3.topics.partitions")
+  public void updatePartitionCount(
+      @Suspended AsyncResponse asyncResponse,
+      @PathParam("clusterId") String clusterId,
+      @PathParam("topicName") String topicName,
+      @Valid PartitionCountRequest partitionCount) {
+
+    if (partitionCount == null) {
+      throw Errors.invalidPayloadException("Request body is empty. Partition_count is required.");
+    }
+
+    CompletableFuture<GetTopicResponse> response =
+        topicManager
+            .get()
+            .updateTopicPartitionCount(topicName, partitionCount.getPartitionCount())
+            .thenCompose(
+                nothing ->
+                    topicManager
+                        .get()
+                        .getTopic(clusterId, topicName)
+                        .thenApply(topic -> GetTopicResponse.create(toTopicData(topic.get()))));
```

Comment:
nit: For the record I am OK with having a check here for the topic not existing race condition - there is a theoretical case that it would address. Feel free to ignore as this is being fast-tracked, so I don't want to request unwanted changes that aren't too important.